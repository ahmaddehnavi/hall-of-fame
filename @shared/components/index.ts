export * from './layout/Col'
export * from './layout/Row'
export * from './screen/Screen'
export * from './backhandler/MultiBackHandler'
export * from './touchable/Touchable'
export * from './button/Button'
export * from './list/FlatList'
export * from './empty/EmptyView'
export * from './loading/LoadingView'
export * from './error/ErrorView'
export * from './async-view/AsyncView'