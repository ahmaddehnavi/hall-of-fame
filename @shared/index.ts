export * from './services'
export * from './utils'
export * from './components'
export * from './modules'
export * from './stores'
