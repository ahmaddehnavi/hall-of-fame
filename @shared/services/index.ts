export * from './base'
export * from './navigation'
export * from './theme'

