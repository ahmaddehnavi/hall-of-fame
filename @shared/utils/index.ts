export * from './di/DIUtil'
export * from './async/AsyncUtil'