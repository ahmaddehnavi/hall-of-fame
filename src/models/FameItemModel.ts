export interface FameItemModel {
    id: string
    name: string
    photo: string
}