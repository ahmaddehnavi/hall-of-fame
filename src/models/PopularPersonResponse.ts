export interface KnownFor {
    original_name: string;
    id: number;
    media_type: string;
    name: string;
    vote_count: number;
    vote_average: number;
    poster_path: string;
    first_air_date: string;
    popularity: number;
    genre_ids: number[];
    original_language: string;
    backdrop_path: string;
    overview: string;
    origin_country: string[];
    video?: boolean;
    title: string;
    original_title: string;
    adult?: boolean;
    release_date: string;
}

export interface PopularPersonItem {
    popularity: number;
    id: number|string;
    profile_path: string;
    name: string;
    known_for: KnownFor[];
    adult: boolean;
}

export interface PopularPersonResponse {
    page: number;
    total_results: number;
    total_pages: number;
    results: PopularPersonItem[];
}