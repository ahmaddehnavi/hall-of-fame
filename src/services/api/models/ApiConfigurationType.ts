export type ApiConfigurationImages = {
    base_url: string;
    secure_base_url: string;
    backdrop_sizes: string[];
    logo_sizes: string[];
    poster_sizes: string[];
    profile_sizes: string[];
    still_sizes: string[];
}

export type ApiConfigurationType = {
    images: ApiConfigurationImages;
    change_keys: string[];
}
