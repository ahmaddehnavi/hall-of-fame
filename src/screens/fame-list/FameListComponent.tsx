import {FlatList, ListResource, Screen} from '@shared';
import autobind from 'autobind-decorator';
import {toJS} from 'mobx';
import {observer} from 'mobx-react';
import React, {Component} from 'react';
import {Image, StyleSheet} from 'react-native';
import {PopularPersonItemType} from '../../services/api/models/PopularPersonResponseType';

type FameListComponentProps = {
    listResource: ListResource<{}, PopularPersonItemType>
    resolveProfileImageUrl: (path: string) => string
}

@observer
export class FameListComponent extends Component<FameListComponentProps> {
    render() {
        let items = toJS(this.props.listResource.items) || [];
        if (items.length) {
            let sheldonCooper: PopularPersonItemType = {
                profile_path: 'https://i.pinimg.com/originals/2e/29/c4/2e29c41787d04c4b3de4aa3832566357.jpg',
                adult: false,
                id: 'special',
                known_for: [],
                name: '',
                popularity: 0
            };
            // insert  sheldon cooper into 3 position
            items.splice(Math.min(2, items.length), 0, sheldonCooper);
        }
        return (
            <Screen style={styles.container}>
                <FlatList
                    style={{
                        flex: 1
                    }}
                    contentContainerStyle={{
                        flexGrow: 1
                    }}
                    resource={this.props.listResource}
                    data={items}
                    renderItem={this.renderItem}
                    removeClippedSubviews
                />
            </Screen>
        )

    }

    @autobind
    renderItem({item, index}: { item: PopularPersonItemType, index: number }) {
        return (
            <Image
                key={item.id}
                source={{uri: this.props.resolveProfileImageUrl(item.profile_path)}}
                resizeMode={'cover'}
                style={styles.image}
            />
        )
    }
}

const styles = StyleSheet.create({
    container: {},
    image: {
        marginBottom: 16,
        marginHorizontal: '5%',
        width: '100%',
        alignSelf: 'center',
        height: 200,
        backgroundColor: '#919191'
    }
});
