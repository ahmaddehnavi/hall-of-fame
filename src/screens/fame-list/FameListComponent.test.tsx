import {ListResource} from '@shared';
import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {PopularPersonItemType} from '../../services/api/models/PopularPersonResponseType';
import {FameListComponent} from './FameListComponent';

describe('FameListComponent Tests', () => {

    jest.useFakeTimers();

    it('renders correctly', () => {
        let res = ListResource.form<{}, PopularPersonItemType>((req, {page}) => {
            return Promise.resolve({
                page: 1,
                isFinished: true,
                items: [{
                    id: 1,
                    adult: false,
                    known_for: [],
                    name: 'test',
                    popularity: 1,
                    profile_path: 'https://test.com/test.png'
                }],
                message: ''
            });
        });

        function resolveImagePath(path) {
            return path;
        }

        let tree = renderer.create(
            <FameListComponent
                listResource={res}
                resolveProfileImageUrl={resolveImagePath}
            />
        );


        expect(tree).toMatchSnapshot();
    });

});
