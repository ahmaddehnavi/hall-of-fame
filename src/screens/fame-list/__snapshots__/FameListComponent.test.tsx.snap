// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FameListComponent Tests renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255,255,255,.5)",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
    <Text
      style={
        Object {
          "marginStart": 8,
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;
